app/models/concerns/foreman_ansible/has_many_ansible_roles.rb

```diff
@@ -6,7 +6,9 @@ module HasManyAnsibleRoles
 
     included do
       def all_ansible_roles
-        (ansible_roles + inherited_ansible_roles).uniq
+        result = (ansible_roles + inherited_ansible_roles).uniq
+        result += host_ansible_roles if is_a? Hostgroup
```

Comment:
How about just overriding this method in the hostgroup? Also I think you need to call uniq here to avoid duplicates

```
def all_ansible_roles
  (super += host_ansible_roles).uniq
end
```